BootloaderCommonPkg/Library/Ext23Lib/Ext2Fs.c

```diff
@@ -277,12 +277,23 @@ ReadInode (
   UINT32        RSize;
   INT32         Rc;
   DADDRESS      InodeSector;
+  EXT2GD       *Ext2FsGrpDes;
   EXTFS_DINODE *DInodePtr;
 
   Fp = (FILE *)File->FileSystemSpecificData;
   FileSystem = Fp->SuperBlockPtr;
 
-  InodeSector = FSBTODB (FileSystem, INODETOFSBA (FileSystem, INumber));
+  Ext2FsGrpDes = FileSystem->Ext2FsGrpDes;
+  Ext2FsGrpDes = (EXT2GD*)((UINT32)Ext2FsGrpDes + (INOTOCG(FileSystem, INumber) * FileSystem->Ext2FsGDSize));
+
+  InodeSector = Ext2FsGrpDes->Ext2BGDInodeTables + (((INumber - 1) % FileSystem->Ext2Fs.Ext2FsINodesPerGroup) / FileSystem->Ext2FsInodesPerBlock);
```

Comment:
better to use ModU64x32 and DivU64x32 instead.  It will also help check  divided by 0 issue.

Reply:
Updated it as advised. I might need to do another patch to clean up the types/sizes being used to truly make it 64bit and follow edk2 styles.